contracts/CodexRecordFees.sol

```diff
@@ -32,25 +33,34 @@ contract CodexRecordFees is Pausable {
   // Fee to modify tokens. 10^18 = 1 token
   uint256 public modificationFee = 0;
 
-  // Parameters in the polynomial used to calculate discount
-  // Current range of discount is 0-100% (i.e., stake enough tokens for a 100% discount!)
-  uint256 public lowerBound = 0;
-  uint256 public upperBound = 100;
+  // To receive a full discount on the protocol fees, stake this number of tokens
+  uint256 public tokensNeededForFullDiscount;
 
   modifier canPayFees(uint256 baseFee) {
     if (feeRecipient != address(0)) {
-      // @TODO: Update the discount to be based on weight as opposed to just
-      //  a binary on/off value
       uint256 calculatedFee = baseFee;
-      if (codexStakeContainer != address(0) &&
-        codexStakeContainer.totalStakedFor(msg.sender) > 0) {
 
-        calculatedFee = 0;
+      if (codexStakeContainer != address(0) && tokensNeededForFullDiscount > 0) {
+        uint256 totalStakedFor = codexStakeContainer.totalStakedFor(msg.sender);
+
+        if (totalStakedFor > 0) {
+          // Since floating point operations aren't supported in the EVM, we need to use a divisor to simulate fractions
+          uint256 divisor = 1 ether;
+
+          // The calculated discount is a % of tokens staked against the tokens needed to be staked
+          // e.g., if tokensNeededForFullDiscount is 100, and totalStakedFor is 50, the calculated discount is 50%
+          uint256 calculatedDiscount = totalStakedFor.mul(divisor).div(tokensNeededForFullDiscount);
```

Comment:
I need to write a test case for this, but I think there's a logic bug here that would cause the contract to actually transfer tokens back to the user if they stake enough (i.e., a negative fee!).

Reply:
This was indeed an issue but it's fixed now.